azure/functions/decorators/dapr_function_app.py

```diff
@@ -0,0 +1,509 @@
+                        name=arg_name,
+                        pub_sub_name=pub_sub_name,
+                        topic=topic,
+                        dapr_address=dapr_address,
+                        data_type=parse_singular_param_to_enum(data_type,
+                                                               DataType),
+                        **kwargs))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def dapr_binding_output(self,
+                            arg_name: str,
+                            binding_name: str,
+                            operation: str,
+                            dapr_address: Optional[str] = None,
+                            data_type: Optional[
+                                Union[DataType, str]] = None,
+                            **kwargs) \
+            -> Callable[..., Any]:
+        """The dapr_binding_output decorator adds
+        :class:`DaprBindingOutput` to the :class:`FunctionBuilder` object
+        for building :class:`Function` object used in worker function
+        indexing model.
+        This is equivalent to defining DaprBindingOutput
+        in the function.json which enables function to send a value to
+        a Dapr output binding.
+        All optional fields will be given default value by function host when
+        they are parsed by function host.
+
+        Ref: https://aka.ms/azure-function-dapr-binding-output-binding
+
+        :param arg_name: The name of the variable that represents DaprState
+        output object in function code.
+        :param arg_name: The name of the variable that represents DaprState
+        input object in function code.
+        :param binding_name: The configured name of the binding.
+        :param operation:  The configured operation.
+        :param dapr_address: Dapr address, it is optional field, by default
+        it will be set to http://localhost:{daprHttpPort}.
+        :param data_type: Defines how Functions runtime should treat the
+        parameter value
+        :param kwargs: Keyword arguments for specifying additional binding
+        fields to include in the binding json
+
+        :return: Decorator function.
+        """
+
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_binding(
+                    binding=DaprBindingOutput(
+                        name=arg_name,
+                        binding_name=binding_name,
+                        operation=operation,
+                        dapr_address=dapr_address,
+                        data_type=parse_singular_param_to_enum(data_type,
+                                                               DataType),
+                        **kwargs))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+
+class DaprFunctionApp(FunctionRegister, DaprTriggerApi, DaprBindingApi):
+
+    def __init__(self,
```

Comment:
You should add more documentation for this - as a way to market DAPR as well.

<img width="781" alt="image" src="https://github.com/Azure/azure-functions-python-library/assets/1047040/495c1b06-793f-4059-a997-f52301d23ce7">